src/components/attendee/NavOverlay/ScheduleOverlay/ScheduleOverlay.tsx

```diff
@@ -0,0 +1,196 @@
+        );
+
+        const personalisedSchedule = eventsFilledWithPriority.filter(
+          (event) => event.isSaved
+        );
+        const formattedDay = formatDayLabel(day);
+
+        return {
+          scheduleDate: formattedDay,
+          daysEvents: showPersonalisedSchedule
+            ? personalisedSchedule
+            : eventsFilledWithPriority,
+        };
+      })
+      .filter((scheduleDay) => !!scheduleDay.daysEvents.length);
+  }, [
+    isScheduleTimeshifted,
+    dayDifference,
+    liveAndFutureEvents,
+    firstScheduleDate,
+    showPersonalisedSchedule,
+  ]);
+
+  const bookmarkEvent = useCallback(
+    (
+      e: React.MouseEvent<HTMLDivElement, MouseEvent>,
+      event: ScheduledEvent
+    ) => {
+      if (!userWithId?.id) return;
+
+      // @debt get rid of stopPropagation() in the project allowing a valid event bubbling
+      e.stopPropagation();
+
+      event.isSaved
+        ? removeEventFromPersonalizedSchedule({ event, userId: userWithId.id })
+        : addEventToPersonalizedSchedule({ event, userId: userWithId.id });
+    },
+    [userWithId?.id]
+  );
+
+  const renderedEvents = useMemo(
+    () =>
+      weekdays.map(({ scheduleDate, daysEvents }) => (
+        <div key={scheduleDate}>
+          <div>{scheduleDate}</div>
+          {daysEvents.map((event) => {
+            const isCurrentEventLive = isEventLive(event);
+            const showDate = Boolean(
+              differenceInCalendarDays(
+                eventEndTime({ event }),
+                eventStartTime({ event })
+              )
+            );
+
+            return (
+              <div key={event.id}>
+                <div>{event.name}</div>
+                <div>
+                  <span>
+                    {isCurrentEventLive
+                      ? "Live "
+                      : formatTimeLocalised(eventStartTime({ event })) + " - "}
+                  </span>
+                  <span>{isCurrentEventLive && "until "}</span>
+                  <span>
+                    {showDate &&
+                      formatDateRelativeToNow(eventEndTime({ event }))}
+                  </span>{" "}
+                  <span>{formatTimeLocalised(eventEndTime({ event }))}</span>
+                  <span> in {event.spaceId}</span>
+                </div>
+                <div>{event.description}</div>
+                <div
+                  className={styles.ScheduleOverlay__bookmark}
+                  onClick={(e) => bookmarkEvent(e, event)}
+                >
+                  {event.isSaved
+                    ? "Removed from bookmarked"
+                    : "Bookmark this event"}
+                </div>
+              </div>
+            );
+          })}
+        </div>
+      )),
+    [weekdays, bookmarkEvent]
+  );
+
+  return (
+    <div className={styles.ScheduleOverlay__wrapper}>
+      <div className={styles.ScheduleOverlay__header}>
+        {space?.name ?? SPACE_TAXON.title} schedule
```

Comment:
What if the space name is empty string?

Reply:
not going to happen ( space form validation prevents that )

Reply:
Until someone changes the validation in the future :) 
That's the problem with TS, you tell it is a string, and `""` is valid string. It will not complain